File: apps/web/src/features/work-orders/utils/inProgressPartyLabel.ts
import type { InProgressWorkOrderItem } from '../types/work-order.types';

export function getInProgressPartyLabel(item: InProgressWorkOrderItem): string {
  if (item.owner?.fullName) {
    return item.owner.fullName;
  }
  if (item.broughtByName) {
    return `Traído por ${item.broughtByName}`;
  }
  return 'Sin propietario';
}

File: apps/android/app/src/main/java/com/mecatrack/mobile/domain/InProgressPartyLabel.kt
package com.mecatrack.mobile.domain

fun inProgressPartyLabel(ownerFullName: String?, broughtByName: String?): String {
    val owner = ownerFullName?.trim().orEmpty()
    if (owner.isNotEmpty()) {
        return owner
    }
    val broughtBy = broughtByName?.trim().orEmpty()
    if (broughtBy.isNotEmpty()) {
        return "Traído por $broughtBy"
    }
    return "Sin propietario"
}
